src/app.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Flask application which receives GitHub webhooks and logs those."""
+
+import json
+import logging
+import os
+import sys
+from pathlib import Path
+from typing import Any
+
+from flask import Flask, request
+
+app = Flask(__name__)
+app.config.from_prefixed_env()
+
+
+def _log_file_name() -> str:
+    """Create the log file name.
+
+    Returns:
+        The log file name.
+    """
+    # We use the process ID to avoid race conditions between multiple instances of the app.
+    pid = os.getpid()
+    return f"webhooks.{pid}.log"
```

Comment:
Maybe it will be good adding more uniqueness to the log filename, as the process ID can be reused, which may result in the file being overwritten.
```python
datetime.now().strftime(f"webhooks.%Y-%m-%d-%H-%M-%S.{pid}.logs")
```